site/content/en/references/kustomize/kustomization/buildmetadata/_index.md

```diff
@@ -0,0 +1,308 @@
+
+- `managedByLabel`
+- `originAnnotations`
+- `transformerAnnotations`
+
+It is possible to set one or all of these options in the kustomization file:
+
+It is possible for to set both `transformerAnnotations` and `originAnnotations`:
+
+```yaml
+buildMetadata: [managedByLabel, originAnnotations, transformerAnnotations]
+```
+
+### Managed By Label
+To mark the resource as having been managed by kustomize, you can specify the `managedByLabel`
+option in the `buildMetadata` field of the kustomization:
+
+```yaml
+buildMetadata: [managedByLabel]
+```
+
+This will add a label `app.kubernetes.io/managed-by` to each resource with the version of kustomize 
+that has built it. For example, given the following input:
+
+kustomization.yaml
+```yaml
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+resources:
+- service.yaml
+buildMetadata: [managedByLabel]
+```
+
+service.yaml
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: myService
+spec:
+  ports:
+  - port: 7002
+```
+
+`kustomize build` will produce a resource with an output like the following:
+
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    app.kubernetes.io/managed-by: kustomize-v444.333.222
+  name: myService
+spec:
+  ports:
+  - port: 7002
+```
+
+
+### Origin Annotation
+To retain the information about the origin of a resource, you can specify the `originAnnotations` 
+option in the `buildMetadata` field of the kustomization: 
+
+```yaml
+buildMetadata: [originAnnotations]
+```
+
+When this option is set, generated resources should receive an annotation with key `config.kubernetes.io/origin`,
+containing data about the generator that produced it. If the resource is from the `resources` field, this annotation
+contains data about the file it originated from. 
+
+The possible fields of these annotations are:
+
+```yaml
+config.kubernetes.io/origin: |
+  path: path.yaml # The path to the resource file itself
+  ref: v0.0.0 # If from a remote file or generator, the ref of the repo URL
+  repo: http://github.com/examplerepo # If from a remote file or generator, the repo source
+  configuredIn: path/to/generatorconfig # If a generated resource, the path to the generator config
+  configuredBy: # If a generated resource, the ObjectReference of the generator config
+    kind: Generator
+    apiVersion: builtin 
+    name: foo
+    namespace: default
+```
+
+All local file paths are relative to the top-level kustomization, i.e. the kustomization file in the directory upon 
+which `kustomize build` was invoked. For example, if someone were to run `kustomize build foo`, all file paths 
+in the annotation output would be relative to`foo/kustomization.yaml`. 
+
+All remote file paths are relative to the root of the remote repository. 
+
+Any fields that are not applicable would be omitted from the final output. If a generator is invoked via 
+a field in the kustomization file, `configuredIn` would point to the kustomization file itself. 
```

Comment:
If you split out the field list, I suggest also moving this to the field explanation itself.